### ISSUE TITLE
Assistance with the commands I need to run in creating custom image for raspi-ltsp

### ISSUE THREAD

**Issue by greavette** [maxnet/berryboot#117]
Hello,

I'm testing out a new raspberry pi distro called raspi-ltsp.  It creates an ltsp server as well as the necessary boot files for each Raspberry Pi to connect to the ltsp server.  I know there is already berryterminal but I'm testing out creating an image for raspi-ltsp for berryboot as well.

The raspi-ltsp creates a folder on the ltsp server called piboot.  You simply copy these files to a formatted SD card for the raspberry pi to use.  Everyything in raspi-ltsp runs in ram and doesn't actually run from the sd card so I don't think there is a second partitiion to mount and squash when creating the image.  So when I run the following command in creating the berryboot image:
sudo kpartx -av raspiltsp.img 
What I see is only:
add map loop0p1 (252:0): 0 7618560 linear /dev/loop0 8192

The 'adding custom distributions' page suggests I'll see two loop0p1 and loop0p2.
http://www.berryterminal.com/doku.php/berryboot/adding_custom_distributions

It would seem that some of the commands to add custom distributions aren't needed.  What would I have to run to get my image built for raspi-ltsp?  

Here is a page describing the raspi-ltsp project:
http://gbaman.github.io/RaspberryPi-LTSP/

Any help or direction you can provide would be greatly appreciated.

**Comment by maxnet** [comment]
Reason you only have one partition is because it mounts the root file system (normally the second partition) from a network share.
Try converting the contents of /opt/ltsp/armhf on your LTSP server with mksquashfs.

**Comment by greavette** [comment]
Thank you for the reply Max.  Do I need to convert the contents of the /opt/ltsp/armhf for berryboot to work with connecting to this ltsp server?  

I guess what I'm seeing is all I need is one partition for raspi-ltsp to work from the SD Card.  The benefit to using berryboot would be to easily update each thin client SD Card using the option in berryboot to mount a new image from network.  I'd rather keep the raspi-ltsp to only use the files from piboot on the sd card.  But how do I do this?  Does berryboot need to have two partitions?  If not, how do I add the files I need to an image file in berryboot to make this connection work to raspi-ltsp?

Thank you.

**Comment by maxnet** [comment]
Mounting a root file system off a network share is not a feature easily available in the Berryboot GUI.

If you want to go that way you will have to write a shell script that manually mounts the NFS share, switches to the root file system and executes the init script.
Something along the lines of:

```
mkdir -p /tmp/remote-mountpoint
mount 10.0.0.1:/opt/ltsp/armhf /tmp/remote-mountpoint
exec switch_root /tmp/remote-mountpoint /sbin/init-ltsp
```

(assuming NFS is used, might need different commands for NBD)
Use "berryboot-init" as the name of the shell script, and convert it to squashfs format with mksquashfs.

In short: Berryboot supports two kinds of SquashFS image files:
- Ones that have the full root file system of an operating system in them. Which is what you get if you where to convert the contents of the second partition of a SD card image to SquashFS format. In this case it would be the contents of  /opt/ltsp/armhf 
- Ones that only have a file named "berryboot-init" in them, which will be executed as shell script by Berryboot. You can use this for special cases. E.g. to mount a network share, and execute something from there.

**Comment by greavette** [comment]
Not sure I totally understand all you are explaining to me...but if it makes any difference, the raspi-ltsp system is a fat client that runs completelyin RAM.  The files on the SD card have all they need to connect to the ltsp server. So I think what you are saying though is using berryboot I would need a special script to mount a network share.  This is more work than I thought it would take.  

I think I'll work instead to use berryterminal to connect to the raspi-ltsp server instead.  For now when I try berryterminal I can see the ltsp login screen, it lets me login with no error, but then all I see is a completely black screen with just my cursor moving around...no desktop.  I'll open an issue on berryterminal to ask what would be causing this to occur.

Thanks Max.

**Comment by maxnet** [comment]
> the raspi-ltsp system is a fat client that runs completely in RAM.

Seems to me raspi-ltsp is connecting to a NFS or NBD network share.

https://github.com/gbaman/RaspberryPi-LTSP/blob/master/boot/cmdline.txt

```
init=/sbin/init-ltsp nbdroot=1.1.1.1:/opt/ltsp/armhf
```

Means: connect to NBD network share /opt/ltsp/armhf on server 1.1.1.1, and start the init script /sbin/init-ltsp you will find there.

Like I mentioned, if you want to replicate those cmdline.txt settings in Berryboot, you will need to create a shell script that does that.

> I think I'll work instead to use berryterminal to connect to the raspi-ltsp server instead. For now when I try 
> berryterminal I can see the ltsp login screen, it lets me login with no error, but then all I see is a 
> completely black screen with just my cursor moving around...no desktop. I'll open an issue on 
> berryterminal to ask what would be causing this to occur.

Make sure you are using Edubuntu 12.x or 13.x.
Not 14.x, that is broken.

**Comment by greavette** [comment]
Thanks for the comments.  You are right, it seems raspi-ltsp does connect via NBD network share.

As for using LTSP 14.x, I did it get working with Berryterminal finally.  I took your suggestion to install LXDE on my LTSP server and update my cmdline.txt file to use lxsession.  I've been able to connect to my ltsp server from berryterminal finally.  Lots more testing for me to do but so far so good.

Thanks!
